_blog/2023-12-20-loco-rust-rails.mdx

```diff
@@ -0,0 +1,219 @@
+}
+```
+
+We can extend the behaviour of our ActiveModel by adding a `before_save()` method as mentioned before, like so:
+```rust
+use sea_orm::entity::prelude::{ActiveModelBehavior};
+
+use super::_entities::notes::ActiveModel;
+
+impl ActiveModelBehavior for ActiveModel {
+    // extend activemodel below (keep comment for generators)
+    async fn before_save<C>(self, _db: &C, insert: bool) -> Result<Self, DbErr>
+    where
+        C: ConnectionTrait,
+        {
+            println!("This is happening before we save something!");
+            Ok(self)
+    }
+}
+```
+The `ActiveModelBehaviour` trait implementation (from `sea_orm`) allows us to define behaviour for an `ActiveModel` - more specifically, we can add methods for before and after saving a model, as well as before and after deleting a model. We can also extend the behaviour of our model by adding extra methods to it:
+
+```rust
+impl super::_entities::users::Model {
+    // .. your own methods
+}
+```
+
+Now we can use it in a handler function by loading the item from the database - then we can do whatever we need to with the data:
+```rust
+async fn load_item(ctx: &AppContext, id: i32) -> Result<Model> {
+    let item = Entity::find_by_id(id).one(&ctx.db).await?;
+    item.ok_or_else(|| Error::NotFound)
+}
+
+pub async fn update(
+    Path(id): Path<i32>,
+    State(ctx): State<AppContext>,
+    Json(params): Json<Params>,
+) -> Result<Json<Model>> {
+    // use sea_orm to load an item based on the id
+    let item = load_item(&ctx, id).await?;
+    
+    // turn the item into an ActiveModel that we can then use
+    let mut item = item.into_active_model();
+    
+    // update the parameters of the current item with the new properties
+    params.update(&mut item);
+    
+    // feed the new item back into the database
+    let item = item.update(&ctx.db).await?;
+    
+    // return the updated item
+    format::json(item)
+}
+```
+
+However - that isn't all that Loco.rs has to offer. We can also use the `loco_rs` Validator struct to be able to verify a new model before needing to do anything with it! A use case for this, for example, might be if we needed to check if an email is a valid email. You can check this out below:
+```rust
+[derive(Debug, Validate, Deserialize)]
+pub struct ModelValidator {
+    #[validate(length(min = 2, message = "Name must be at least 2 characters long."))]
+    pub name: String,
+    #[validate(custom = "validation::is_valid_email")]
+    pub email: String,
+}
+
+impl From<&ActiveModel> for ModelValidator {
+    fn from(value: &ActiveModel) -> Self {
+        Self {
+            name: value.name.as_ref().to_string(),
+            email: value.email.as_ref().to_string(),
+        }
+    }
+}
+```
+
+## Job Processing
+Like with everything else in Loco, you can also generate workers and tasks via the CLI. Running `cargo run generate task` or `cargo run generate worker` will let you generate a task or a worker at will.
```

Comment:
it's probably good to go over the doc and replace `cargo run` with `cargo loco`. `loco` is a cargo alias to save a bit more typing (which you get for free in your generated project)